File: bible-tools/src/main/java/rawcomposition/bibletools/info/api/BibleToolsApi.java
package rawcomposition.bibletools.info.api;

import rawcomposition.bibletools.info.model.json.ReferencesResponse;
import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;


/**
 * Created by tinashe on 2015/02/15.
 */
public interface BibleToolsApi {

    @GET("/resources/get/{book}/{chapter}/{verse}")
    Observable<ReferencesResponse> deliverReferences(
            @Path("book") int book,
            @Path("chapter") int chapter,
            @Path("verse") int verse
    );
}

File: bible-tools/src/main/java/rawcomposition/bibletools/info/data/retrofit/BibleToolsApi.kt
package rawcomposition.bibletools.info.data.retrofit

import io.reactivex.Completable
import io.reactivex.Observable
import rawcomposition.bibletools.info.data.model.Reference
import rawcomposition.bibletools.info.data.model.StrongsResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path


interface BibleToolsApi {

    @GET("/api/v1.0/verse/{ref}")
    fun getReference(@Path("ref") ref: String): Observable<Response<Reference>>

    @POST("/api/v1.0/helpful/{resource_id}")
    fun submitHelpful(@Path("resource_id") resId: String): Completable

    @POST("/api/v1.0/unhelpful/{resource_id}")
    fun submitUnHelpful(@Path("resource_id") resId: String): Completable

    @GET("/api/v1.0/word/{word_id}")
    fun getStrongs(@Path("word_id") wordId: String): Observable<Response<StrongsResponse>>
}
